qutebrowser/mainwindow/statusbar/backforward.py

```diff
@@ -0,0 +1,42 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2017 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Navigation (back/forward) indicator displayed in the statusbar."""
+
+from PyQt5.QtCore import pyqtSlot
+from qutebrowser.mainwindow.statusbar import textbase
+from qutebrowser.utils import log
+
+
+class Backforward(textbase.TextBase):
+
+    """Shows navigation indicator (if you can go backward and/or forward)."""
+
+    def on_tab_changed(self, tab):
+        self.tab = tab
```

Comment:
I'm not entirely happy with saving the tab here, but it's a bit of a tricky situation - what about connecting the signal above like `tabs.cur_url_changed.connect(functools.partial(status.backforward.on_url_changed, tabs=tabs))`, then taking a `tabs` in `on_url_changed`, and use `tabs.currentWidget()` to get the current tab? I haven't tested it, but I think it should work and it seems less error-prone to me.